OrcScala/src/orc/Backend.scala

```diff
@@ -80,13 +90,26 @@ trait Compiler[+CompiledCode] {
     override def toURL = toURI.toURL
   }
 
+  class ManyCompilationExceptions(exceptions: Seq[Throwable]) extends CompilationException("Multiple compiler errors") {
+    exceptions.foreach(addSuppressed)
+  }
+
   /** Compile the code in the reader using the given options and produce error messages on the
     * err writer. This is a simple wrapper around the other compile function.
     */
-  @throws(classOf[IOException])
+  @throws(classOf[IOException]) @throws(classOf[CompilationException])
```

Comment:
I have addressed this issue by having two different entry points. The exception throwing one is not used anywhere at the moment. We can revisit using it for testing later (to get better error messages than "compile failed" or similar).